src/sql/mod.rs

```diff
@@ -25,8 +25,8 @@ pub(crate) enum Part {
 }
 
 impl SqlBuilder {
-    pub(crate) fn new(template: &str) -> Self {
-        let mut iter = template.split('?');
+    pub(crate) fn new(template: &str, arg: &str) -> Self {
```

Comment:
Instead of changing the signature, you can add a [second constructor](https://rust-lang.github.io/api-guidelines/naming.html#casing-conforms-to-rfc-430-c-case) instead - it's more idiomatic:

```rust
pub(crate) fn new(template: &str) -> Self { 
    Self::with_arg(template, "?")
}

pub(crate) fn with_arg(template: &str, arg: &str) -> Self;
```

This way you don't have to manually update all these tests.

Reply:
since this isn't a public API adding param is fine